pylink/unlockers/unlock_idcode.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2025 Jeremiah Gillis
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .. import enums
+from .. import errors
+
+import ctypes
+
+# Global variable to store the ID code
+global_id_code = None
```

Comment:
A global won't play well with multiple instances (e.g. if someone is debugging two chips at the same time). It looks like we need it for the callback in the dialog. Could we pass a partial function instead? That would allow us to prepopulate the `id_code`:

```
import functools


def set_unloock_idcode(self, id_code):
    partial = functools.partial(unlock_idcode_hook_dialog, id_code)
    unlock_idcode_cb = enums.JLinkFunctions.UNLOCK_IDCODE_HOOK_PROTOTYPE(partial)
```